R/model_LaTeX.R

```diff
@@ -26,42 +26,37 @@ generate_latex <- function(equations) {
   ))
 }
 
-SIR_LaTeX <- function(mu) {
-  switch(mu + 1, 
-         helpText(generate_latex(c(SIR_nonVD_equation))),
-         helpText(generate_latex(c(SIR_VD_equation)))
-  )
+SIR_LaTeX <- function(mu,qValue) {
+  forceOfInfection <- getForceOfInfection(qValue)
+  helpText(generate_latex(SIR(mu,forceOfInfection)))
 }
 
-SIRS_LaTeX <- function(mu) {
-  switch(mu + 1, 
-         helpText(generate_latex(c(SIRS_nonVD_equation))),
-         helpText(generate_latex(c(SIRS_VD_equation )))
-  )
+SIRS_LaTeX <- function(mu,qValue) {
+  forceOfInfection <- getForceOfInfection(qValue)
+  helpText(generate_latex(SIRS(mu,forceOfInfection)))
 }
 
-SIRD_LaTeX <- function(mu) {
-  switch(mu + 1,
-         helpText(generate_latex(c(SIRD_nonVD_equation))),
-        helpText(generate_latex(c(SIRD_VD_equation)))
-  )
+SIRD_LaTeX <- function(mu,qValue) {
+  forceOfInfection <- getForceOfInfection(qValue)
+  helpText(generate_latex(SIRD(mu,forceOfInfection)))
 }
 
-SEIR_LaTeX <- function(mu) {
-  switch(mu + 1,
-         helpText(generate_latex(c(SEIR_nonVD_equation))),
-         helpText(generate_latex(c(SEIR_VD_equation)))
-  )
+SEIR_LaTeX <- function(mu,qValue) {
+  forceOfInfection <- getForceOfInfection(qValue)
+  helpText(generate_latex(SEIR(mu,forceOfInfection)))
 }
 
-SEIRD_LaTeX <- function(mu) {
-  switch(mu + 1,
-         helpText(generate_latex(c(SEIRD_nonVD_equation))),
-         helpText(generate_latex(c(SEIRD_VD_equation)))
-  )
+SEIRS_LaTeX <- function(mu,qValue) {
+  forceOfInfection <- getForceOfInfection(qValue)
+  helpText(generate_latex(SEIRS(mu,forceOfInfection)))
+}
+
+SEIRD_LaTeX <- function(mu,qValue) {
+  forceOfInfection <- getForceOfInfection(qValue)
+  helpText(generate_latex(SEIRD(mu,forceOfInfection)))
 }
 
 ## FIXME: is this working or not?
-renderModelLaTeX <- function(model, vitalStatistics) {
-  tagList(withMathJax(eval(call(paste0(model, "_LaTeX"), vitalStatistics))))
+renderModelLaTeX <- function(model, vitalStatistics,massAction) {
+  tagList(withMathJax(eval(call(paste0(model, "_LaTeX"), vitalStatistics, massAction))))
```

Comment:
Can you integrate the _effect_ of calling each of these into the `renderModelLaTeX` function?

As every function is so similar (which is a good sign, you did well throughout this PR), it will be more maintainable (by enabling us to make the same change in one place rather than six) if you put the two assignment statements into the mentioned function.

The `vitalStatistics` and `qValue` areguments you need and pass along are already available.

Please modify the function I mentioned; you'll need to read about `eval` and `call` a little, but otherwise this PR is great.

Reply:
Actually, because it's easier than a back-and-forth and I know what needs to change, here's the change. Please integrate it and then remove the other changes after verifying the function arguments are what are expected.
```r
renderModelLaTeX <- function(model, vitalStatistics, massAction) {
  eval(call(model, vitalStatistics, getForceOfInfection(massAction)))
  helpText(generate_latex())
```